extension-manifest-v3/src/utils/engines.js

```diff
@@ -142,6 +142,10 @@ async function loadFromStorage(name) {
       return engine;
     }
   } catch (e) {
+    // If there is an error loading the engine from storage, the DB must be corrupted.
+    // In this case, we should delete it, as it will be reloaded on the next run.
+    IDB.deleteDB('engines');
```

Comment:
To avoid races, I would recommend to wait for the purge operation to finish.

```suggestion
    await IDB.deleteDB('engines').catch((e2) => console.error('Failed to cleanup corrupted engine db', e2));
```

Reply:
Sure, good point.